src/did/did-resolver.ts

```diff
@@ -1,12 +1,14 @@
 import type { PublicJwk } from '../jose/types';
 import { Did } from './did';
+import { MemoryCache } from '../utils/memory-cache';
 
 /**
  * TODO: add docs, Issue #72 https://github.com/TBD54566975/dwn-sdk-js/issues/72
  */
 export class DidResolver {
   didResolvers: Map<string, DidMethodResolver>;
 
+  memoryCache = new MemoryCache(86400);//Time to live in seconds is 24 hours
   // TODO: add DIDCache to constructor method signature, Issue #62 https://github.com/TBD54566975/dwn-sdk-js/issues/62
   constructor(resolvers: DidMethodResolver[]) {
```

Comment:
Pass in the cache through constructor as an optional parameter such that it can be changed and mocked. e.g.

```suggestion
  constructor(resolvers: DidMethodResolver[], cache?: Cache) {
```

Also the TODO should be removed since this PR addresses the issue.

Reply:
sorry dont understand this, though i did add the cache as an optional parameter however what do you mean by dependency on interface

Reply:
> what do you mean by dependency on interface

The DidResolver should have no reference/knowledge of "memory cache" except for the construction in constructor.

> also what operations are to be done on the cache passed in?

Not sure what you mean.

Reply:
ok thanks

Reply:
the thing still confusing me is the fact that we have the memoryCache being instantiated but then if we are passing in a cache as an optional parameter to allow for ease of testing what operations does the cache passed in do? For example it cant instantiate the memoryCache class as it requires a parameter of TTL(time to live) meaning in cant modify that class unless im missing something, so essentially what do i do with the cache parameter passed in.

Reply:
You will only instantiate the memory cache if no cache is passed in assuming cache is an optional parameter, in which case you should simply instantiate the cache with a default timeout as suggested. The design of cache as an "optional" parameter is simply a convenience, if the developer doesn't like to use memory cache or the default timeout set for memory cache, fine, pass in their own.

One could argue for making the cache always required to be passed in, I am okay with that, not too opinionated on that, but dependency injection of the cache is a must.

Reply:
ok thanks that makes sense but cba you also check the issues comments as there is a new problem affecting testing of the code

Reply:
can

Reply:
Answered in issue #62

Reply:
i have another question so right now i have modified my code to have a boolean arg which will allow for injection of a new cache and if its false will use memorycache instantiating and youre able to pass in a timeout else default value is 600.

Now the question is if the type of cache can be modified then how will that affect the functions currently being used for example .get() cannot be used if the type of cache is Cache but can be used if its MemoryCache?

Reply:
unless im being confused and cache:Cache is only meant to be used for testing?

Reply:
Cache is the interface. So as long as object of type Cache is passed in, the resolver can use it the same way regardless of implementation.